Objects/genobject.c

```diff
@@ -427,10 +429,12 @@ _gen_throw(PyGenObject *gen, int close_on_genexit,
             tstate->frame = gen->gi_frame;
             /* Close the generator that we are currently iterating with
                'yield from' or awaiting on with 'await'. */
+            PyFrameState state = gen->gi_frame->f_state;
+            gen->gi_frame->f_state = FRAME_EXECUTING;
             ret = _gen_throw((PyGenObject *)yf, close_on_genexit,
                              typ, val, tb);
             tstate->frame = f;
-            gen->gi_running = 0;
+            gen->gi_frame->f_state = state;
```

Comment:
Nit: for symmetry, I think the new modified line would be better before the `tstate->frame = f;` line.

Reply:
done